### PR TITLE
Edit and get dashboard variables

### PR DESCRIPTION
**Pull request by zaher-aa**
## Endpoints
1. `PATCH` | `/api/v1/user/edit-dashboard-vars`
2. `GET` | `/api/v1/user/dashboard-vars`

### Request
```json
{
   "nextupToOwedSplitPercentage": "number"
   "systemActivationDate": "date",
   "fetchMaxCount": "number",
   "expiredAfterInYears": "number",
   "viewliftEmail": "email",
   "viewliftEndpoint": "url",
   "viewliftWatchesFetchLimit": "number",
   "viewliftPassword": "string"
}
```

```json
{
  "message": "DASHBOARD VARS CHANGED SUCCESSFULLY",
  "data": {
    "regularVariables": {
      "nextupToOwedSplitPercentage": 1,
      "systemActivationDate": "2022-02-02T00:00:00.000Z",
      "fetchMaxCount": 10,
      "expiredAfterInYears": 23424,
      "viewliftEmail": "ali@g.com",
      "viewliftEndpoint": "https://owiefj.com",
      "viewliftWatchesFetchLimit": 1000
    },
    "encryptedVariables": {
      "viewliftPassword": "test"
    }
  }
}
```
> Fail status: 400
```json
{
  "message": "SAME OLD PASSWORD"
}
```

## `GET` | `/api/v1/user/dashboard-vars`

```json
{
  "data": {
    "regularVariables": {
      "nextupToOwedSplitPercentage": 1,
      "systemActivationDate": "2022-02-02T00:00:00.000Z",
      "fetchMaxCount": 10,
      "expiredAfterInYears": 23424,
      "viewliftEmail": "ali@g.com",
      "viewliftEndpoint": "https://owiefj.com",
      "viewliftWatchesFetchLimit": 1000
    },
    "encryptedVariables": {
      "viewliftPassword": "wefwef"
    }
  }
}
```

### CODE REVIEW

File: src/helpers/constants.ts [hunk 1]
```diff
@@ -24,6 +24,8 @@ export const messages = {
     resetPassword: 'RESET PASSWORD',
     conflict: 'CONFLICT',
     userStatistics: 'USER STATISTICS',
+    SAME_PASSWORD: 'SAME OLD PASSWORD',
```
Comment (AymanSami98) [inline src/helpers/constants.ts:27]: Why use snake_case with capital letters?

**Review by AymanSami98 — APPROVED** [review]
Great Job

File: src/services/user/editDashboardVars.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+type IEditDashboardVars = (_: IVariables) => Promise<IVariables>
+
+const editDashboardVars: IEditDashboardVars = async ({
+  encryptedVariables,
+  regularVariables,
+}) => {
+  const oldVars = (await Settings.findOne({ where: { name: 'variables' } })) as ISettings;
+  const oldVarsValue = JSON.parse(oldVars.value as string) as IVariables;
+
```
Comment (Ibrahim-Jarada) [inline src/services/user/editDashboardVars.ts:16]: No need to parse it as its type JSON in DataTypes so Sequelize will handle parsing it
```suggestion
  const oldVarsValue = oldVars.value as IVariables;
```

File: src/services/user/getDashboardVars.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import config from '../../config';
+
+const { ENCRYPTION_SECRET_KEY } = config.server;
+
+type IGetDashboardVars = () => Promise<IVariables>
+
+const getDashboardVars: IGetDashboardVars = async () => {
+  const oldVars = (await Settings.findOne({ where: { name: 'variables' } })) as ISettings;
+  const oldVarsValue = JSON.parse(oldVars.value as string) as IVariables;
+
```
Comment (Ibrahim-Jarada) [inline src/services/user/getDashboardVars.ts:12]: Same as above
No need to parse it as its type JSON in DataTypes so Sequelize will handle parsing it
```suggestion
  const oldVarsValue = oldVars.value as IVariables;
```

**Review by Ibrahim-Jarada — APPROVED** [review]

**Review by Ibrahim-Jarada — CHANGES_REQUESTED** [review]
Mainly namings

File: src/helpers/constants.ts (hunk 1 above)
Comment (zaher-aa) [reply src/helpers/constants.ts:27]: Constants should be CAPITALIZED

**Review by Ibrahim-Jarada — APPROVED** [review]

**Pull request merged**